### ISSUE TITLE
renaming OVA/OAD episodes title

### ISSUE THREAD

**Issue by papaj-na-wrotkach** [ShokoAnime/ShokoServer#1040]
> **Anime Entry Types That Typically Only Have One (1) Regular Episode Entry**
> 
> These anime entries typically only have 1 regular episode entry (regular episode type, meaning S episodes are excluded):
> 
> 1. OVA/OAD
> 2. Movie
> 3. Music Video
> 4. Web Series (when used to release an OVA/OAD, Movie, or Music Video; does not apply to Web Series used to release a TV series)
> 
> ***For these anime entries, when it only has 1 regular-type episode entry, the episode title is added to anime entry's title and not the episode entry's title; the entry's single episode title follows default rules as outlined above.***
> 
> 1. Generally, this rule applies only for cases when there are one ([Hajimete no Gal: Hajimete no Bunkasai](https://anidb.net/a13248)) or two ([Koi to Uso: Isshou no Koi / Koi no Kimochi](https://anidb.net/a14024)) episode titles for the single episode release.
> 2. In rare cases when there are more than 2 ([Tottoko Hamtarou: Anime Dechu!](https://anidb.net/a13387)) episode titles, do not add them to the OVA/OAD's anime title; they are added like usual to the episode entry's title.
>
> quoted from: [wiki.anidb.net](https://wiki.anidb.net/Content:Titles#Anime_Entry_Types_That_Typically_Only_Have_One_(1)_Regular_Episode_Entry)

As far as I know, there's no way to change episode title and my single-episode OVA's titles look like this:
![image](https://user-images.githubusercontent.com/29522138/215336447-0585564f-4a66-4199-8189-957d8c871b47.png)

I would like to change `OVA` to series title - in this case to `Teishi Kyoushitsu no Vampire - Oppai, Tsutsumimasu!`.

Additionally the server could automatically rename the episode title in the same way as it renames files.

To achieve what I want the renamers should have access to information about series relations:
| | |
| --- | --- |
| **OVA series name** | Hajimete no Gal: My First School Festival |
| **prequel/base series name** | Hajimete no Gal |
| **new OVA series name** | My First School Festival |

The renaming rules could also be added to season titles. In my case I would like to rename the old `High School DxD New: Teishi Kyoushitsu no Vampire - Oppai, Tsutsumimasu!` to `High School DxD New OVA`. This case also requires the information about series relations. For now I can do it manually but it would be convenient if it was done automatically.

**TL;DR**

Add support for:
- manually renaming episodes (like you can do with series)
- renamers for episode titles (like for file names)
- *renamers for series titles

**Comment by da3dsoul** [comment]
Manually renaming episodes - not high priority, but I'd say worth it to add
Renamers for things that aren't files - it wouldn't be built in, but with the modularity and events for plugins that we are adding bit by bit, it could be done via plugins. I don't think it's worth it to make a whole system for renaming things via script.

**Comment by papaj-na-wrotkach** [comment]
> Manually renaming episodes - not high priority, but I'd say worth it to add

Cool, could you post info on this issue when there will be progress with the functionality?

> Renamers for things that aren't files - it wouldn't be built in, but with the modularity and events for plugins that we are adding bit by bit, it could be done via plugins. I don't think it's worth it to make a whole system for renaming things via script.

Does the plugin system provide everything I need to build such a plugin? I would like to write a plugin if that's possible.

**Comment by da3dsoul** [comment]
> Does the current plugin system provide everything I need to build such a plugin? I would like to write a plugin if that's possible.

Not yet. We are slowly building out the plugin system's capabilities. It's quite new

**Comment by papaj-na-wrotkach** [comment]
Okay, I'll close this issue and open a new one - specifically for the manual episode renaming functionality. Later I'll check what can I write with existing plugin system and if there's something worth adding/re-implementing I'll open another issue.

**Comment by papaj-na-wrotkach** [comment]
where should I open the issue? In this repo or in [ShokoDesktop repo](https://github.com/ShokoAnime/ShokoDesktop)?

**Comment by da3dsoul** [comment]
That would go in Server

**Comment by revam** [comment]
Renamers have access to relations

**Comment by da3dsoul** [comment]
Different relations @revam. Those were song relations, which we don't have
